File: ios/Plugin/UsefilAudioPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(UsefilAudioPlugin, "UsefilAudio",
        CAP_PLUGIN_METHOD(play64, CAPPluginReturnPromise);
        CAP_PLUGIN_METHOD(play, CAPPluginReturnPromise);
        CAP_PLUGIN_METHOD(stop, CAPPluginReturnPromise);
        CAP_PLUGIN_METHOD(preload, CAPPluginReturnPromise);

)

File: ios/Plugin/UsefilAudioPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(UsefilAudioPlugin)
public class UsefilAudioPlugin: CAPPlugin {
    private let implementation = UsefilAudio()

    @objc func play64(_ call: CAPPluginCall) {
        implementation.play64(call)
    }

    @objc func play(_ call: CAPPluginCall) {
        implementation.play(call)
    }
    
    @objc func stop(_ call: CAPPluginCall) {
        implementation.stop(call)
    }
    
    @objc func preload(_ call: CAPPluginCall) {
        implementation.preload(call)
    }
}
